server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -2039,7 +2039,16 @@ private void innerOpenEngineAndTranslog(LongSupplier globalCheckpointSupplier) t
     }
 
     private boolean assertSequenceNumbersInCommit() throws IOException {
-        final Map<String, String> userData = SegmentInfos.readLatestCommit(store.directory()).getUserData();
+        final Map<String, String> userData;
+        if (indexSettings.isRemoteSnapshot() && indexSettings.getExtendedCompatibilitySnapshotVersion() != null) {
+            // Inefficient method to support reading old Lucene indexes
+            userData = Lucene.readSegmentInfosExtendedCompatibility(
+                store.directory(),
+                indexSettings.getExtendedCompatibilitySnapshotVersion()
+            ).getUserData();
+        } else {
+            userData = SegmentInfos.readLatestCommit(store.directory()).getUserData();
+        }
```

Comment:
nit: The code would read a bit better if this were abstracted into a separate function